### PR TITLE
Flatbuffers serialization, aggregation optimization, build improvements

### PR DESCRIPTION
**Pull request by chester-leung**
* Adds integration test for C++ model aggregation
* Replaces existing serialization with Flatbuffers
* Optimizes enclave aggregation code
* Build improvements

Note: I did not include Flatbuffers installation and build as part of this PR. I opened #6 to note this. To install Flatbuffers you can do `snap install flatbuffers`.

Co-authored-by: David Yi < davidyi624@berkeley.edu>
Co-authored-by: Rishabh Poddar <rishabhp89@gmail.com>
Co-authored-by: Octavian Sima <34696537+octaviansima@users.noreply.github.com>

### CODE REVIEW

File: server/host/host.cpp [hunk 1]
```diff
@@ -1,6 +1,4 @@
 #include <openenclave/host.h>
```
Comment (podcastinator) [inline server/host/host.cpp:1]: File permissions to 644

File: server/host/host.cpp [hunk 2]
```diff
@@ -1,6 +1,4 @@
...
-#include <stdio.h>
-#include <iostream>
 
 #include "enclave.h"
```
Comment (podcastinator) [inline server/host/host.cpp:3]: This file `enclave.h` seems to be empty. Perhaps an oversight? I believe that's where the `Enclave` class was defined.

File: server/host/host.cpp [hunk 3]
```diff
@@ -14,22 +12,32 @@ using namespace std;
 char* path = "./enclave/enclave.signed";
 uint32_t flags = OE_ENCLAVE_FLAG_DEBUG | OE_ENCLAVE_FLAG_SIMULATE;
 
+// Helper function used to copy double pointers from untrusted memory to enclave memory
```
Comment (podcastinator) [inline server/host/host.cpp:15]: This function is not being used.

**Review by podcastinator — CHANGES_REQUESTED** [review]
Thanks! Left some comments. In addition, there are a few empty files. Please delete them if not required?

File: common/utils.h [hunk 4]
```diff
@@ -1,8 +1,10 @@
-#include <stdio.h>
+#ifndef _UTILS_H
```
[the remark this answers is not in the document]
Comment (chester-leung) [reply common/utils.h:1]: fixed

File: common/encryption/encrypt.h [hunk 5]
```diff
@@ -1,44 +1,18 @@
 #ifndef ENCRYPT_H_
```
[the remark this answers is not in the document]
Comment (chester-leung) [reply common/encryption/encrypt.h:1]: Fixed

File: server/enclave/ecalls.cpp [hunk 6]
```diff
@@ -27,77 +27,73 @@ if (!oe_is_outside_enclave((ptr), size)) {                \
 }
```
[the remark this answers is not in the document]
Comment (chester-leung) [reply server/enclave/ecalls.cpp:27]: fixed

File: server/host/host.cpp (hunk 1 above)
Comment (chester-leung) [reply server/host/host.cpp:1]: Fixed

File: server/host/host.cpp (hunk 2 above)
Comment (chester-leung) [reply server/host/host.cpp:3]: Oh interesting, looks like the code in that file wasn't pushed for whatever reason. Thanks for catching this!

File: server/host/host.cpp (hunk 3 above)
Comment (chester-leung) [reply server/host/host.cpp:15]: Removed

File: common/encryption/encrypt.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (chester-leung) [reply common/encryption/encrypt.h:1]: The other files in this directory also seemed to have strange permissions, I changed them all back to 644.

**Review by podcastinator — APPROVED** [review]

**Pull request merged**